File: frontend/src/lib/api/evalSession.ts
import createApiClient from "@/lib/axios";

const API_BASE = process.env.NEXT_PUBLIC_API_URL || "http://localhost:8001";
const api = createApiClient(API_BASE);

export interface EvalSessionDetail {
  id: string;
  created_at: string;
  updated_at: string;
  phone: string;
  profile_id: string;
  profile_name: string;
  current_step: string;
  data: Record<string, any>;
  totals: Record<string, any>;
}

export const fetchLatestActiveEvalSessionByProfile = async (
  profile_id: string
): Promise<{ session: EvalSessionDetail | null }> => {
  const { data } = await api.get("/api/eval-sessions/latest-active/by-profile", {
    params: { profile_id },
  });
  return data;
};

export const createEvalSession = async (payload: {
  phone: string;
  profile_id: string;
  profile_name: string;
}) => {
  const { data } = await api.post("/api/eval-sessions", payload);
  return data;
};

export const patchEvalSessionQuiet = async (
  session_id: string | null,
  payload: {
    current_step?: string;
    patch?: Record<string, any>;
    append_llm?: Record<string, any>;
  }
) => {
  if (!session_id) return null;
  
  // Transform payload to match backend EvalSessionPatch
  const body = {
    current_step: payload.current_step,
    data: payload.patch, // assuming patch contains the updated data
  };

  const { data } = await api.post(`/api/eval-sessions/${session_id}/update`, body);
  return data;
};

File: backend/app/schemas/eval_session.py
from pydantic import BaseModel
from typing import Optional, Dict, Any

class EvalSessionCreate(BaseModel):
    phone: Optional[str] = ""
    profile_id: str
    profile_name: Optional[str] = ""

class EvalSessionPatch(BaseModel):
    current_step: Optional[str] = None
    data: Optional[Dict[str, Any]] = None
    totals: Optional[Dict[str, Any]] = None

class EvalSessionSummaryOut(BaseModel):
    id: str
    created_at: str
    updated_at: str
    phone: str
    profile_id: str
    profile_name: str
    current_step: str
    data: Optional[Dict[str, Any]] = None
    totals: Optional[Dict[str, Any]] = None
